prog/victoria_metrics/victoria_metrics_resource_nexus.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+require_relative "../../lib/util"
+
+class Prog::VictoriaMetrics::VictoriaMetricsResourceNexus < Prog::Base
+  subject_is :victoria_metrics_resource
+
+  def self.assemble(project_id, name, location_id, admin_user, admin_password, storage_size_gib, node_count, vm_size)
+    project = Project[project_id]
+    fail "No existing project" unless project
+
+    location = Location[location_id]
+    fail "No existing location" unless location
+
+    unless node_count == 1
+      fail "Only single node is supported"
+    end
+
+    Validation.validate_vm_size(vm_size, "x64")
+    Validation.validate_name(name)
+    Validation.validate_minio_username(admin_user)
+
+    DB.transaction do
+      ubid = VictoriaMetricsResource.generate_ubid
+      root_cert_1, root_cert_key_1 = Util.create_root_certificate(common_name: "#{ubid} Root Certificate Authority", duration: 60 * 60 * 24 * 365 * 5)
+      root_cert_2, root_cert_key_2 = Util.create_root_certificate(common_name: "#{ubid} Root Certificate Authority", duration: 60 * 60 * 24 * 365 * 5)
+
+      victoria_metrics_resource = VictoriaMetricsResource.create(
+        name: name,
+        location_id: location.id,
+        admin_user: admin_user,
+        admin_password: admin_password,
```

Comment:
We can generate admin_password instead of accepting as a parameter, similar to MinIO and Postgres.